source/WeightedList.h

```diff
@@ -69,20 +75,75 @@ class WeightedList {
 
 private:
 	std::vector<Type> choices;
+	std::vector<std::size_t> weights;
 	std::size_t total = 0;
 };
 
 
 
+template <class T>
+std::size_t erase(WeightedList<T> &list, const T &item)
```

Comment:
You can implement `erase` by calling `erase_if` with a predicate that simply compares to `item`.